src/Pact/Native.hs

```diff
@@ -963,6 +974,20 @@ where' i as@[k',app@TApp{},r'] = gasUnreduced i as $ ((,) <$> reduce k' <*> redu
   _ -> argsError' i as
 where' i as = argsError' i as
 
+distinct :: GasRNativeFun e
+distinct g i = \case
+    [l@(TList v _ _ )] | V.null v -> pure (g,l)
+    [TList{..}] -> _tList
+        & V.toList
+        & L.nubBy termEq
+        & V.fromList
+        & toTListV _tListType def
+        & pure
+        & computeGas' g i (GDistinct $ square $ V.length _tList)
+    as -> argsError i as
+  where
+    square = (^ (2 :: Int))
```

Comment:
I don't think costing in n^2 is necessary since `GDistinct` is already similar in complexity to `GSort` and you've costed it the same with the `expLengthPenalty`.

Reply:
done